datasets/narrativeqa_manual/narrativeqa_manual.py

```diff
@@ -0,0 +1,342 @@
+    "summaries": "https://raw.githubusercontent.com/deepmind/narrativeqa/master/third_party/wikipedia/summaries.csv",
+    "qaps": "https://raw.githubusercontent.com/deepmind/narrativeqa/master/qaps.csv",
+}
+
+
+class NarrativeQaManual(datasets.GeneratorBasedBuilder):
+    """The NarrativeQA Manual dataset"""
+
+    VERSION = datasets.Version("1.0.0")
+
+    @property
+    def manual_download_instructions(self):
+        return """ You need to manually download the stories for this dataset using the script provided by the authors \
+                (https://github.com/deepmind/narrativeqa/blob/master/download_stories.sh). Running the shell script creates a folder named "tmp"\
+                in the root directory and downloads the stories there. This folder containing the stories\
+                can be used to load the dataset via `datasets.load_dataset("narrativeqa_manual", data_dir="<path/to/folder>")."""
+
+    def _info(self):
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "kind": datasets.Value("string"),
+                    "story_url": datasets.Value("string"),
+                    "story_file_size": datasets.Value("string"),
+                    "wiki_url": datasets.Value("string"),
+                    "wiki_title": datasets.Value("string"),
+                    "story_word_count": datasets.Value("string"),
+                    "story_start": datasets.Value("string"),
+                    "story_end": datasets.Value("string"),
+                    "story_text": datasets.Value("string"),
+                    "summary": datasets.Value("string"),
+                    "summary_tokenized": datasets.Value("string"),
+                    "qaps": datasets.features.Sequence(
+                        {
+                            "question": datasets.Value("string"),
+                            "answer1": datasets.Value("string"),
+                            "answer2": datasets.Value("string"),
+                            "question_tokenized": datasets.Value("string"),
+                            "answer1_tokenized": datasets.Value("string"),
+                            "answer2_tokenized": datasets.Value("string"),
+                        }
+                    ),
+                }
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URLS)
+        documents = data_dir["documents"]
+        summaries = data_dir["summaries"]
+        qaps = data_dir["qaps"]
+
+        train_dict = {}
+        test_dict = {}
+        valid_dict = {}
+        with open(documents, encoding="utf-8") as csv_file:
```

Comment:
`_split_generators` is not supposed to read the data files. Its role is to define the splits and provide them with the necessary resources (i.e. downloaded data files).

Can you move all this processing to `_generate_examples ` instead ?
In the `gen_kwargs` defined for each split you can simply pass the directories:
```python
return [
            datasets.SplitGenerator(
                name=datasets.Split.TRAIN,
                gen_kwargs={
                    "data_dir": data_dir,
                    "manual_dir": manual_dir,
                    "split": "train",
                },
            ),
            datasets.SplitGenerator(
                name=datasets.Split.TEST,
                gen_kwargs={
                    "data_dir": data_dir,
                    "manual_dir": manual_dir,
                    "split": "test",
                },
            ),
            datasets.SplitGenerator(
                name=datasets.Split.VALIDATION,
                gen_kwargs={
                    "data_dir": data_dir,
                    "manual_dir": manual_dir,
                    "split": "valid",
                },
            ),
        ]
```
And then when generating the examples you can ignore the examples that don't belong to the split that is being generated
```python
            for id_, row in enumerate(csv_reader):
                if row:
                    if row[1] != split:
                        continue
```